### DISCUSSION TITLE
HttpCode for /auth/email/register

### DISCUSSION THREAD

**Discussion by LucaAndrei** [brocoders/nestjs-boilerplate#1346] [Q&A] [answered]
Currently the HttpCode is set to @HttpCode(HttpStatus.NO_CONTENT)
Wouldn't it be better to set it to @HttpCode(HttpStatus.CREATED) ?

**Comment by Shchepotin** [comment, marked as answer]
@LucaAndrei reason why we use HttpStatus.NO_CONTENT is we not returned payload for `/auth/email/register`. Why do we not return the payload: because we handle 2 scenarios:

1. We disallow login users without email confirmation
1. We allow login users without email confirmation

For the first scenario, we shouldn't return any payload about a user within the register flow.

If you have any thoughts about it, please share we consider it.
